Brokerages/Alpaca/AlpacaBrokerage.cs

```diff
@@ -0,0 +1,449 @@
+                throw;
+            }
+
+        }
+
+        /// <summary>
+        /// Places a new order and assigns a new broker ID to the order
+        /// </summary>
+        /// <param name="order">The order to be placed</param>
+        /// <returns>True if the request for a new order has been placed, false otherwise</returns>
+        public override bool PlaceOrder(Order order)
+        {
+            const int orderFee = 0;
+            order.PriceCurrency = "USD";
+
+            lock (_locker)
+            {
+                try
+                {
+                    var apOrder = GenerateAndPlaceOrder(order);
+                    order.BrokerId.Add(apOrder.OrderId.ToString());
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e);
+                    if (e.InnerException != null)
+                        OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, 100, e.InnerException.Message));
+                    return false;
+                }
+
+            }
+            OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee) { Status = Orders.OrderStatus.Submitted });
+
+            return true;
+        }
+
+        /// <summary>
+        /// Updates the order with the same id
+        /// </summary>
+        /// <param name="order">The new order information</param>
+        /// <returns>True if the request was made for the order to be updated, false otherwise</returns>
+        public override bool UpdateOrder(Order order)
+        {
+            return false;
+        }
+
+        /// <summary>
+        /// Cancels the order with the specified ID
+        /// </summary>
+        /// <param name="order">The order to cancel</param>
+        /// <returns>True if the request was made for the order to be canceled, false otherwise</returns>
+        public override bool CancelOrder(Order order)
+        {
+            Log.Trace("AlpacaBrokerage.CancelOrder(): " + order);
+
+            if (!order.BrokerId.Any())
+            {
+                Log.Trace("AlpacaBrokerage.CancelOrder(): Unable to cancel order without BrokerId.");
+                return false;
+            }
+
+            foreach (var orderId in order.BrokerId)
+            {
+                CheckRateLimiting();
+                var task = _restClient.DeleteOrderAsync(new Guid(orderId));
+                task.SynchronouslyAwaitTaskResult();
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Gets the history for the requested security
+        /// </summary>
+        /// <param name="request">The historical data request</param>
+        /// <returns>An enumerable of bars covering the span specified in the request</returns>
+        public override IEnumerable<BaseData> GetHistory(HistoryRequest request)
+        {
+            var exchangeTimeZone = _marketHours.GetExchangeHours(Market.USA, request.Symbol, request.Symbol.SecurityType).TimeZone;
```

Comment:
I think we cannot use `_symbolExchangeTimeZones` here because it is only populated and used in the Alpaca `DataQueueHandler` (which would not be used in the cloud).
Also `GetHistory` could be called by external downloaders.